docs/tutorial.md

```diff
@@ -0,0 +1,274 @@
+bot.craft(plankRecipe, 1);
+const stickRecipe = bot.recipesFor(280); // 280 ID for Stick
+bot.craft(plankRecipe, 1);
+```
+
+Correct approach with callback:
+
+```js
+const plankRecipe = bot.recipesFor(5)[0]; // 5 ID for Oak Wood Planks
+bot.craft(plankRecipe, 1, null, function (err) {
+  if (err) return bot.chat(err.message);
+  const stickRecipe = bot.recipesFor(280); // 280 ID for Stick
+  bot.craft(plankRecipe, 1, null, function (err) {
+    bot.chat("Crafting Stick finished");
+  });
+});
+```
+
+More readable version
+
+```js
+const plankRecipe = bot.recipesFor(5)[0]; // 5 ID for Oak Wood Planks
+
+function stickCrafted(err) {
+  if (err) return bot.chat(err.message);
+  bot.chat("Crafting Stick finished");
+}
+
+function plankCrafted(err) {
+  if (err) return bot.chat(err.message);
+  const stickRecipe = bot.recipesFor(280)[0]; // 280 ID for Stick
+  bot.craft(plankRecipe, 1, null, stickCrafted);
+}
+
+bot.craft(plankRecipe, 1, null, plankCrafted);
+```
+
+#### Consuming Bot
+
+In this example we create bot that eat / consume held item if hungry.
+Using
+[`health`](http://mineflayer.prismarine.js.org/#/api?id=health) event of the bot to listen on any changes in health or food points.
+[`bot.consume`](http://mineflayer.prismarine.js.org/#/api?id=botconsumecallback) method to eat.
+[`bot.food`](http://mineflayer.prismarine.js.org/#/api?id=botfood) property to check food points.
+`bot.heldItem` property to get the name of the item held by the bot.
+
+```js
+var mineflayer = require("mineflayer");
+var bot = mineflayer.createBot({...})
+
+const minFoodPoints = 18;
+
+bot.on("health", function () {
+  // listen on any changes in health or food points.
+  if (bot.food < minFoodPoints) {
+    // if the food points lower than 18, then
+    var heldItemName = bot.heldItem.name;
+
+    function onFinishEating(err){
+      if (err) return bot.chat(err.message);
+      bot.chat(`I'm finish eating ${heldItemName}`);
+    }
+
+    bot.chat(`Eating ${heldItemName}`);
+    bot.consume(onFinishEating);
+    // `consume` method takes `onFinishEating` as callback, and execute `onFinishEating` when its finish or stumble an error.
+  }
+})
+```
+
+#### Fisherman Bot
+
+Can be found in [fishing.js](https://github.com/PrismarineJS/mineflayer/blob/master/examples/fisherman.js#L52).
+
+## FAQ
+
+### How to create discord bot with mineflayer
+
+Answer in [discord.js](https://github.com/PrismarineJS/mineflayer/blob/master/examples/discord.js)
+
+### How to run bot on android
+
+Here quick setup for running bot on android device using [Termux](https://termux.com/).
+
+#### Install Termux
+
+Install [Termux](https://termux.com/) and start it.
+
+#### Setup
+
+First, update the Termux packages
+
+```bash
+pkg update -y
+```
+
+Now install `nodejs`, `npm` should be automaticaly installed as well:
+
+```bash
+pkg install nodejs -y
+```
+
+Finally, install `mineflayer` library:
+
+```bash
+npm install mineflayer@latest
+```
```

Comment:
You've made an assumption that the user knows how to use npm and install mineflayer at the beginning of the document

Reply:
At the beginning of basics section not in faq.

Reply:
This is the same step. You can explain how to install nodejs in termux and just say installation and usage of modules is the same or explain how to install mineflayer at the beginning